### PR TITLE
JS-166 The AST should be serialized and sent to the plugin only when at least one consumer is listening

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**

### CODE REVIEW

File: packages/jsts/src/analysis/analyzer.ts [hunk 1]
```diff
@@ -86,12 +86,18 @@ function analyzeFile(
...
-    return {
+    const result: JsTsAnalysisOutput = {
       issues,
       ucfgPaths,
       ...extendedMetrics,
-      ast: serializeAst(sourceCode, filePath),
     };
+
+    const ast = serializeAst(sourceCode, filePath, input.skipAst);
+    if (ast) {
```
Comment (quentin-jaquier-sonarsource) [inline packages/jsts/src/analysis/analyzer.ts:96]: We can probably avoid the extra argument, and check the value of `input.skipAst` inside the if condition instead.

File: sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/BridgeServer.java [hunk 2]
```diff
@@ -19,6 +19,8 @@
  */
 package org.sonar.plugins.javascript.bridge;
 
+import static org.sonarsource.api.sonarlint.SonarLintSide.INSTANCE;
+
```
Comment (quentin-jaquier-sonarsource) [inline sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/BridgeServer.java:23]: Is this a specific setup for your repo? It does not follow the layout defined in the [common toolbox](https://github.com/SonarSource/sonar-developer-toolset):

1. import all other imports
2. "blank line"
3. import static all other imports

File: sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/analysis/AnalysisConsumers.java [hunk 3]
```diff
@@ -54,4 +54,8 @@ public void accept(JsFile jsFile) {
   public void doneAnalysis() {
     consumers.forEach(JsAnalysisConsumer::doneAnalysis);
   }
+
+  public int getSize() {
+    return consumers.size();
+  }
```
Comment (quentin-jaquier-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/analysis/AnalysisConsumers.java:60]: We can replace this with "HasConsumer" and the body with `!isEmpty`.

File: sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/BridgeServer.java (hunk 2 above)
Comment (ilia-kebets-sonarsource) [reply sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/BridgeServer.java:23]: My intellij must be misconfigured.

**Review by quentin-jaquier-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**